components/brave_wallet_ui/components/desktop/popup-modals/filter-modals/filter-components/filter-dropdown-section.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Dropdown from '@brave/leo/react/dropdown'
+
+// Utils
+import {
+  getLocale
+} from '../../../../../../common/locale'
+
+// Types
+import {
+  DropdownFilterOption
+} from '../../../../../constants/types'
+
+// Styled Components
+import {
+  CheckboxText,
+  Description,
+  IconWrapper,
+  Icon,
+  DropdownWrapper
+} from './filter-components.style'
+import {
+  Row,
+  Column
+} from '../../../../shared/style'
+
+interface Props {
+  title: string
+  description: string
+  icon?: string
+  dropdownOptions: DropdownFilterOption[]
+  selectedOptionId: string
+  onSelectOption: (id: string) => void
+}
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'dropdown-option': React.HTMLAttributes<HTMLElement> & {
+        value?: string
+        children?: any
+        key?: React.Key
+      }
+    }
+  }
+}
```

Comment:
please move this to a declarations file.

maybe `declarations/elements.d.ts`?

Reply:
Added an issue to remove this and use the provided `leo-option` once a `key` prop is added.
https://github.com/brave/brave-browser/issues/30826